src/wp-includes/blocks.php

```diff
@@ -1263,9 +1263,9 @@ function build_query_vars_from_query_block( $block, $page ) {
 			}
 		}
 		if ( isset( $block->context['query']['sticky'] ) && ! empty( $block->context['query']['sticky'] ) ) {
-			$sticky = get_option( 'sticky_posts' );
+			$sticky = (array) get_option( 'sticky_posts' );
 			if ( 'only' === $block->context['query']['sticky'] ) {
-				$query['post__in'] = $sticky;
+				$query['post__in'] = $sticky ? $sticky : array( 0 );
```

Comment:
```suggestion
				$query['post__in'] = $sticky ? $sticky : array( 0 );
				$query['ignore_sticky_posts'] = 1;
```

I think to make this work we also have to set `ignore_sticky_posts` as [done in Gutenberg](https://github.com/WordPress/gutenberg/blob/ce49c68c128616b51df78f7443114fe6b6ddd972/lib/compat/wordpress-6.1/blocks.php#L180).

Reply:
Can we also add the comment?
```
/**
 * Passing an empty array to post__in will return have_posts() as true (and all posts will be returned).
 * Logic should be used before hand to determine if WP_Query should be used in the event that the array
 * being passed to post__in is empty.
 *
 * @see https://core.trac.wordpress.org/ticket/28099
 */
```

Reply:
```suggestion
				/**
				 * Passing an empty array to post__in will return have_posts() as true (and all posts will be returned).
				 * Logic should be used before hand to determine if WP_Query should be used in the event that the array
				 * being passed to post__in is empty.
				 *
				 * @see https://core.trac.wordpress.org/ticket/28099
				 */
				$query['post__in']            = ! empty( $sticky ) ? $sticky : array( 0 );
				$query['ignore_sticky_posts'] = 1;
```

This is a full backport of the changes in Gutenberg https://github.com/WordPress/gutenberg/blob/42dad571eac32c41a7ff34f0da80fcc4e5ceb3bd/lib/compat/wordpress-6.1/blocks.php#L171-L183.

Reply:
huh, the batch commit didn't pick up all of the changes.

Reply:
@ocean90 I hesitated adding ignore_sticky_posts thinking there might be no reason for it without verifying that assumption.  +1 for adding it :)